graylog2-server/src/test/java/org/graylog/testing/completebackend/apis/Search.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.testing.completebackend.apis;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.collect.ImmutableSet;
+import io.restassured.path.json.JsonPath;
+import io.restassured.specification.RequestSpecification;
+import org.bson.types.ObjectId;
+import org.graylog.plugins.views.search.elasticsearch.ElasticsearchQueryString;
+import org.graylog.plugins.views.search.rest.QueryDTO;
+import org.graylog.plugins.views.search.rest.SearchDTO;
+import org.graylog.plugins.views.search.searchtypes.MessageList;
+import org.graylog2.plugin.indexer.searches.timeranges.AbsoluteRange;
+import org.graylog2.plugin.indexer.searches.timeranges.InvalidRangeParametersException;
+import org.graylog2.plugin.indexer.searches.timeranges.TimeRange;
+import org.graylog2.shared.bindings.providers.ObjectMapperProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.notNullValue;
+
+
+public class Search implements GraylogRestApi {
+
+    public List<String> searchForAllMessages() {
+        List<String> messages = new ArrayList<>();
+
+        waitFor(() -> captureMessages(messages::addAll), "Timed out waiting for messages to be present");
+
+        return messages;
+    }
+
+    public boolean waitForMessage(String message) {
+        waitFor(() -> captureMessage(message), "Timed out waiting for message to be present");
+        return true;
```

Comment:
Why does it return boolean?
It is always true, anyway...
I'd rather return "this" to allow chaining of "waitForMessage" invocations.

Reply:
Good point, implemented! Thx!